### ISSUE TITLE
evaluate to include a 3rd math-lib

### ISSUE THREAD

**Issue by nimo23** [ta4j/ta4j#77]
When I have this:

`SimpleLinearRegressionIndicator reg = new SimpleLinearRegressionIndicator(close, 7);`

and it outputs this for index **n**:

```
closeprice(n): "27.82"
simpleLinearReg(n): "27.7761"
```

**does this mean the following ?:**

For index **n** the value of **n+1** is probably around "27.7761" (= **prediction value**).

**Does "SimpleLinearRegressionIndicator" outputs the "predictionValue" for the next comming index?**

Is this something like this from commons-math "SimpleLinearRegression":
http://commons.apache.org/proper/commons-math/userguide/stat.html

```
System.out.println(regression.predict(1.5d))
// displays predicted y value for x = 1.5
```

**Comment by team172011** [comment]
Yes, if you take a look at the test file:

```
    @Test
    public void calculateLinearRegression() {
        double[] values = new double[] { 1, 2, 1.3, 3.75, 2.25 };
        ClosePriceIndicator indicator = new ClosePriceIndicator(new MockTimeSeries(values));
        SimpleLinearRegressionIndicator reg = new SimpleLinearRegressionIndicator(indicator, 5);
        
        SimpleRegression origReg = buildSimpleRegression(values);
        assertDecimalEquals(reg.getValue(4), origReg.predict(4));
    }
```

In the last line the test should be successful if `SimpleLinearRegressionIndicator ` returns value of `origReg(int i)` call from `org.apache.commons.math3.stat.regression.SimpleRegression` that 

> Returns the "predicted" y value associated with the supplied x value, based on the data that has been added to the model when this method is activated.

I think `SimpleLinearRegressionIndicator` should not predict any values, but calculate the current value of the linear regression



> other math tools better from 3rd lib
You are right, the `org.apache.commons.math3` rep is very powerful. We could use it to improve peformance and correctness, but would break with "Minimal 3rd party dependencies" and the "MIT license". Depending on how many indicators could be improved i could imagine to do so.

Regarding also #75

**Comment by nimo23** [comment]
We should evaluate to provide a 3rd party math tools for making all the typical or complex math calculations instead of reinventing the wheel. This would lead to faster and better development without the need to test the complex math calculations..

For example, I want to use the **Kalman-Filter** for our new "ConvergenceDivergenceIndicator.java". 
ConvergenceDivergenceIndicator can, for example, be used for "pair-trading" (Cointegration). Kalman-Filter can be used for predictions, which is very cool.

https://github.com/lukstei/trading-backtest
http://intelligenttradingtech.blogspot.de/2010/05/kalman-filter-for-financial-time-series.html
https://www.quantopian.com/lectures/kalman-filters#watch

So I have to investigate some math libs:

https://statr.me/2015/09/an-overview-of-linear-algebra-libraries-in-scala-java/

and it seems commons-math is the best of all and it has Kalman-Filter, different LinearRegressionModels and a few MachineLearning-Tools. The only thing is: all these libs uses double and ours uses (Big)Decimal. So I dont know if this is a issue for our calculations when including one of those libs.

Of course, we can should have a "KalmanFilterIndicator.java" which I want to wish as a feature request. But the calculations of that within calculate() can be with the help "commons-math" or "la4j".

**Comment by team172011** [comment]
Wont fix because of changes resulting from #75 and #88 (Decimal Interface..)
